From 87c064d37d50800188769573ec3b13a64b6f7b4f Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 23 Mar 2014 14:25:36 +0000
Subject: [PATCH 8/8] PlaylistEditorView: make sure we only update the images
 once in itemVisible

If no image could be found for a node each time a node becomes visible we
would retry to find one. This uses a text field as a flag to make sure we
only look for an image the first time a node becomes visible.
---
 mythplugins/mythmusic/mythmusic/playlisteditorview.cpp |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp b/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
index df5a60e..4617e60 100644
--- a/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
+++ b/mythplugins/mythmusic/mythmusic/playlisteditorview.cpp
@@ -837,8 +837,10 @@ void PlaylistEditorView::treeItemVisible(MythUIButtonListItem *item)
     if (!mnode)
         return;
 
-    if (item->GetImageFilename().isEmpty())
+    if (item->GetText("*").isEmpty())
     {
+        item->SetText(" ", "*");
+
         QString artFile;
 
         if (mnode->getAction() == "trackid")
-- 
1.7.10.2

